metric/system/process/process_windows.go

```diff
@@ -94,6 +94,10 @@ func FetchNumThreads(pid int) (int, error) {
 	if err != nil {
 		return 0, fmt.Errorf("PssCaptureSnapshot failed: %w", err)
 	}
+	defer func() {
+		cHandle, _ := syscall.GetCurrentProcess()
+		PssFreeSnapshot(cHandle, snapshotHandle)
```

Comment:
I think the errors should be returned to caller of FetchNumThreads. If I were using this as a library I would want to know that `PssFreeSnapshot` is failing.

Reply:
thanks, I was going back and forth over that.  changed to return error codes from GetCurrentProcess and PssFreeSnapshot